jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/ImmutableHttpFields.java

```diff
@@ -171,4 +178,75 @@ public String toString()
     {
         return asString();
     }
+
+    private class Listerator implements ListIterator<HttpField>
+    {
+        int _index;
+        int _last = -1;
+
+        Listerator(int index)
+        {
+            _index = index;
+        }
+
+        @Override
+        public void add(HttpField field)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public boolean hasNext()
+        {
+            return _index != _size;
+        }
+
+        @Override
+        public boolean hasPrevious()
+        {
+            return _index > 0;
+        }
+
+        @Override
+        public HttpField next()
+        {
+            if (_index >= _size)
```

Comment:
To be consistent with `previous()` I'd check here for `==`.
Or both get the inequality operators.

Reply:
There was no change here?

Reply:
It is not really symmetric.  It  is plausible that _size might shrink or that index is miscalculated, so index can plausibly be larger than size.   But at the other end, a negative size or index is not really plausible.   But can change previous to `<= 0`.